### DISCUSSION TITLE
How to make the position of nodes randomly generated after the world resets?

### DISCUSSION THREAD

**Discussion by xiaoxiao-ji** [cyberbotics/webots#4514] [Q&A]
System
- Operating System: Windows 10
- Version: Webots R2021b
- Language: Python

I wrote a custom gym environment for the NAO robot according to the .\Webots\projects\samples\howto\openai_gym using the Supervisor class, I want to accomplish random generation of the position of NAO robot after the environment is reset to facilitate the training of the subsequent agent. But the problem I have now is that once the .wbt file in webots is saved, the position of the NAO robot is fixed. I don't know what methods or statements should be used to reset the gym environment to make the location of NAO node change accordingly.

**Comment by ad-daniel** [comment]
You can use the supervisor to move objects/robots around, the process is explained in the first two sections of this tutorial: https://cyberbotics.com/doc/guide/tutorial-8-the-supervisor

**Reply by xiaoxiao-ji** [reply]
Thank you very much for your advice, the problem has been solved. 
But I find a new problem: when I used the supervisor to move NAO robot and ran the simulation as fast as possible, the robot was unstable and would fall down when changing position. Is there any good way to avoid falling?
